index.js

```diff
@@ -111,15 +112,23 @@ var CloudFlare = PromiseObject.create({
 			},
 			promise: function (attempt) {
 				return new BlueBird(function (resolve, reject) {
+          if($self._auth_type == "token") {
+            var headers = {
+              'Authorization': "Bearer " + $self._key,
```

Comment:
I am not sure how this was working for you but the `var headers` definition should be outside of the if statement.

```javascript
          if($self._auth_type == "token") {
            var headers = {
              'Authorization': "Bearer " + $self._key,
              'Content-Type' : 'application/json'
            }
          } else if ($self.auth_type == "x-auth") {
            var headers = {
              'X-Auth-Key': $self._key,
              'X-Auth-Email': $self._email,
              'Content-Type': $config.contentType
            }
          }
```

Like

```javascript
          var headers;
          if($self._auth_type == "token") {
           headers = {
              'Authorization': "Bearer " + $self._key,
              'Content-Type' : 'application/json'
            };
          } else if ($self.auth_type == "x-auth") {
           headers = {
              'X-Auth-Key': $self._key,
              'X-Auth-Email': $self._email,
              'Content-Type': $config.contentType
            };
          } else {
              throw new Error('Invalid auth_type');
          }
```